source/iface/accessor-concept.rst

```diff
@@ -2,41 +2,400 @@
+the data managed by a :ref:`buffer` or :ref:`image <image>`, they provide
+access to local memory on a :ref:`device`, and they define the requirements
+to memory objects which determine the scheduling of kernels.
+
+A memory object requirement is created when an accessor is constructed,
+unless the accessor is a placeholder in which case the requirement is
+created when the accessor is bound to a command by calling
+``sycl::handler::require()``.
+
+There are several different C++ classes that implement accessors:
+
+* The :ref:`command-accessor` class provides access to data in a
+  :ref:`buffer` from within a command.
+* The :ref:`host_accessor` class provides access to data in a
+  :ref:`buffer` from host code that is outside of a command.
+  These accessors are typically used in application scope.
+* The :ref:`local_accessor` class provides access to device
+  local memory from within a SYCL kernel function.
+* The ``sycl::unsampled_image_accessor`` and
+  ``sycl::sampled_image_accessor`` classes provide access to
+  data in an ``sycl::unsampled_image`` and ``sycl::sampled_image``
+  from within a command.
+* The ``sycl::host_unsampled_image_accessor`` and
+  ``sycl::host_sampled_image_accessor`` classes provide access to
+  data in an ``sycl::unsampled_image`` and ``sycl::sampled_image``
+  from host code that is outside of a command. These accessors
+  are typically used in application scope.
+
+Accessor objects must always be constructed in host code,
+either in command group scope or in application scope. Whether the
+constructor blocks waiting for data to synchronize depends on the
+type of accessor:
+
+* Those accessors which provide access to data within a command do
+  not block. Instead, these accessors define a requirement which
+  influences the scheduling of the command.
+* Those accessors which provide access to data from host code do
+  block until the data is available on the host.
+
+.. warning::
+
+  For those accessors which provide access to data within a command,
+  the member functions which access data should only be called from
+  within the command. Programs which call these member functions
+  from outside of the command are ill formed.
+
+.. seealso:: |SYCL_SPEC_ACCESSORS|
+
+==========
+Data types
+==========
+
+All accessors have a ``DataT`` template parameter which specifies
+the type of each element that the accessor accesses. For
+:ref:`command-accessor` and :ref:`host_accessor`, this type must
+either match the type of each element in the underlying :ref:`buffer`,
+or it must be a ``const`` qualified version of that type.
+
+For the image accessors (:ref:`unsampled-image-accessors`,
+:ref:`sampled-image-accessors`), ``DataT`` must be one of:
+
+* ``sycl::int4`` (``sycl::vec<int32_t, 4>``);
+* ``sycl::uint4`` (``sycl::vec<uint32_t, 4>``);
+* ``sycl::float4`` (``sycl::vec<float, 4>``);
+* ``sycl::half4`` (``sycl::vec<half, 4>``).
+
+See :ref:`local_accessor` page for it's allowed ``DataT`` types.
```

Comment:
```suggestion
See :ref:`local_accessor` for it's allowed ``DataT`` types.
```

Reply:
Fixed in the https://github.com/KhronosGroup/SYCL_Reference/pull/96/commits/7544af73fd933d50fd209c31178a4e59475a7c93.